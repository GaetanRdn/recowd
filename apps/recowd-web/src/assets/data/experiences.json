[
  {
    "client": "CBP",
    "roles": [
      {
        "contractType": "FREELANCE",
        "endDate": null,
        "titre": "Lead développeur front-end",
        "startDate": "2022-03-03"
      },
      {
        "contractType": "CDI ESN",
        "endDate": "2022-03-02",
        "titre": "Lead développeur front-end",
        "startDate": "2021-01-01"
      }
    ],
    "description": "Après 30 ans de vie de la chaîne de gestion de sinistres, il est venu le temps d'une refonte complète",
    "localization": "Saint-Herblain, Pays de Loire, France",
    "technologies": [
      "Angular 14",
      "Nx",
      "Typescript",
      "Sass",
      "Angular Material CDK",
      "ngx-translate",
      "Rxjs",
      "Jest",
      "Storybook",
      "Cypress",
      "Gitlab CI"
    ],
    "tasks": [
      "Proposition et mise en place d'une approche atomic design",
      "Former l'équipe à Angular (présentation, pair-programming, points d'échanges)",
      "Prôner l'usage du TDD",
      "Challenger les maquettes fournies par les UI/UXs",
      "Participer aux recrutements pour diverses équipes",
      "travailler de pair avec la PO afin de prioriser les tâches",
      "Remonter les alertes nécessaires sur l'avancée des devs",
      "POC pour la mise en place de Storybook",
      "POC pour la mise en place de Nx",
      "POC pour la mise en place de tests Cypress pour Storybook"
    ]
  },
  {
    "client": "S2E",
    "localization": "Nantes, Pays de Loire, France",
    "description": "Run sur une application d'épargne salariale",
    "roles": [
      {
        "contractType": "CDI ESN",
        "endDate": "2020-12-31",
        "startDate": "2020-06-01",
        "titre": "Développeur full-stack"
      }
    ],
    "technologies": ["Java 8", "Maven", "JSF2", "Gitlab CI"],
    "tasks": ["Maintenance corrective et évolutive"]
  },
  {
    "client": "norsys",
    "localization": "Nantes, Pays de Loire, France",
    "description": "Certains d'entre nous ce sont retrouvés en inter-contrat durant le premier confinement. Afin de ne pas trop subir et d'essayé d'avoir une vie normale, on a monté une équipe afin de réaliser une application permettant géolocaliser sur une carte les missions de nos collaborateurs",
    "roles": [
      {
        "contractType": "CDI",
        "endDate": "2020-06-01",
        "startDate": "2020-04-01",
        "titre": "Lead développeur"
      }
    ],
    "technologies": [
      "Docker",
      "Angular 9",
      "Gitlab CI",
      "OpenStreetMap",
      "NodeJS",
      "MongoDB"
    ],
    "tasks": ["Maintenance corrective et évolutive"]
  },
  {
    "client": "norsys",
    "localization": "Nantes, Pays de Loire, France",
    "description": "Run sur un site web e-commerce de vente de portes",
    "roles": [
      {
        "contractType": "CDI",
        "endDate": "2020-03-31",
        "startDate": "2019-07-01",
        "titre": "Lead développeur"
      }
    ],
    "technologies": ["PHP", "Symfony", "Docker", "Gitlab CI"],
    "tasks": ["Maintenance corrective et évolutive"]
  },
  {
    "client": "Vif",
    "localization": "La Chapelle/Erdre, Pays de Loire, France",
    "description": "Développement from scratch d'une application web à destination des TPEs pour la gestion de leurs produits, clients, commandes et factures.",
    "roles": [
      {
        "contractType": "CDI ESN",
        "endDate": "2019-06-31",
        "startDate": "2017-12-31",
        "titre": "Développeur full-stack"
      }
    ],
    "technologies": [
      "Angular 7",
      "Angular Material 7",
      "Java 8",
      "maven",
      "tomcat",
      "Gitlab CI"
    ],
    "tasks": [
      "Développement des features",
      "Challenger les maquettes fournies par l'UI/UX",
      "Maintenance corrective et évolutive"
    ]
  },
  {
    "client": "Oui.sncf",
    "localization": "Nantes, Pays de Loire, France",
    "description": "Maintenance évolutive et corrective sur le back-office de l'application web de devis, vente et après vente de la SNCF.",
    "roles": [
      {
        "contractType": "CDI ESN",
        "endDate": "2017-12-31",
        "startDate": "2014-10-01",
        "titre": "Développeur backend-end"
      }
    ],
    "technologies": ["REST", "Java 8", "Git", "Maven", "Tomcat"],
    "tasks": [
      "Développement de features / corrections",
      "Chiffrages",
      "Cadrage technique"
    ]
  }
]
