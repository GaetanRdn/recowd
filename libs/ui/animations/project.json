{
  "projectType": "library",
  "sourceRoot": "libs/ui/animations/src",
  "prefix": "rc",
  "targets": {
    "build": {
      "executor": "@nrwl/angular:ng-packagr-lite",
      "outputs": [
        "dist/libs/ui/animations"
      ],
      "options": {
        "project": "libs/ui/animations/ng-package.json"
      },
      "configurations": {
        "production": {
          "tsConfig": "libs/ui/animations/tsconfig.lib.prod.json"
        },
        "development": {
          "tsConfig": "libs/ui/animations/tsconfig.lib.json"
        }
      },
      "defaultConfiguration": "production"
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": [
          "libs/ui/animations/**/*.ts",
          "libs/ui/animations/**/*.html"
        ]
      }
    }
  },
  "tags": [
    "ui"
  ]
}
