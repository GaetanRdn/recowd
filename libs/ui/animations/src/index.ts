export * from './lib/rotate';
export * from './lib/zoom';
