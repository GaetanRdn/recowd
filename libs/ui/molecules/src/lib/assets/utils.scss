@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
