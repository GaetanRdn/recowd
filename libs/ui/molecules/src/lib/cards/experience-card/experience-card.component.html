<rc-card>
  <rc-card-title>{{ experience.client }}</rc-card-title>
  <rc-card-subtitle>
    <span>{{ intervalDates | experienceDuration }}</span>
    <span>{{ experience.localization }}</span>
  </rc-card-subtitle>
  <rc-card-content>
    <rc-accordion>
      <rc-accordion-bloc>
        <rc-accordion-bloc-header>
          <rc-roles-list [roles]="experience.roles"></rc-roles-list>
        </rc-accordion-bloc-header>
        <rc-accordion-bloc-content>
          <p rcQuote>{{ experience.description }}</p>

          <rc-list horizontal>
            <rc-list-title class="label">Technos</rc-list-title>
            <rc-list-item *ngFor="let techno of experience.technos">{{
              techno
            }}</rc-list-item>
          </rc-list>

          <rc-list class="tasks">
            <rc-list-title>Tâches effectuées</rc-list-title>
            <rc-list-item *ngFor="let task of experience.tasks">{{
              task
            }}</rc-list-item>
          </rc-list>
        </rc-accordion-bloc-content>
      </rc-accordion-bloc>
    </rc-accordion>
  </rc-card-content>
</rc-card>
