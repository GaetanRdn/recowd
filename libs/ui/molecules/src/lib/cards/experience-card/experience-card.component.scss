@use '../../assets/utils';

@mixin rc-experience-card($config) {
  $main-content: utils.map-get-deep($config, typography, body-bold);
  $secondary-content: utils.map-get-deep($config, typography, body-regular);
  $primary: utils.map-get-deep($config, colors, primary);

  rc-experience-card {
    .rc-card-subtitle {
      display: flex;
      flex-flow: column;
      gap: 0.8rem;
    }

    .rc-accordion-bloc-header,
    .rc-accordion-bloc-content-wrapper {
      padding-left: 0;
      padding-right: 0;
    }

    .rc-card-content {
      .rc-accordion-bloc-content-wrapper {
        display: flex;
        flex-flow: column;
        gap: 3rem;
      }
    }

    .label {
      color: darken(map-get($primary, color), 30%);
    }

    .tasks {
      .rc-list-item {
        &::before {
          content: '✔️ ';
        }

        &:first-letter {
          text-transform: capitalize;
        }
      }
    }
  }
}
