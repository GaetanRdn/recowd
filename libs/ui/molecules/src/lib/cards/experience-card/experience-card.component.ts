import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import {
  AccordionModule,
  CardModule,
  ListModule,
  QuoteDirective,
} from '@recowd/ui-atoms';
import { Experience, Role } from '@recowd/models';
import { Nullable, Required } from '@recowd/utility-types';
import { ExperienceDurationPipe, FromToDatePipeModule } from '@recowd/pipes';
import { CommonModule } from '@angular/common';
import { RolesListComponent } from '../../lists/roles-list/roles-list.component';
import { TechnologiesListComponent } from '../../lists/technologies-list/technologies-list.component';

@Component({
  selector: 'rc-experience-card',
  standalone: true,
  imports: [
    CommonModule,
    CardModule,
    AccordionModule,
    ListModule,
    ExperienceDurationPipe,
    FromToDatePipeModule,
    QuoteDirective,
    RolesListComponent,
    TechnologiesListComponent,
  ],
  templateUrl: './experience-card.component.html',
  styleUrls: ['./experience-card.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ExperienceCardComponent {
  @Input() @Required() public experience!: Experience;

  get intervalDates(): { startDate: string; endDate: Nullable<string> } {
    const startDate: string = this.experience.roles
      .map((role: Role) => role.startDate)
      .reduce((previousValue, currentValue) => {
        if (Date.parse(previousValue) < Date.parse(currentValue)) {
          return previousValue;
        }
        return currentValue;
      });

    const endDate: Nullable<string> = this.experience.roles
      .map((role: Role) => role.endDate)
      .reduce((previousValue, currentValue) => {
        if (previousValue === null) {
          return previousValue;
        }

        if (currentValue === null) {
          return currentValue;
        }

        if (Date.parse(previousValue) < Date.parse(currentValue)) {
          return previousValue;
        }
        return currentValue;
      });

    return { startDate, endDate };
  }
}
