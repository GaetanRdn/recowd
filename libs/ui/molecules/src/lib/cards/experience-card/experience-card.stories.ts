import { Meta, moduleMetadata, Story } from '@storybook/angular';
import { ExperienceCardComponent } from './experience-card.component';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

export default {
  title: 'molecules/cards/experience-card',
  decorators: [
    moduleMetadata({
      imports: [ExperienceCardComponent, BrowserAnimationsModule],
    }),
  ],
} as Meta<ExperienceCardComponent>;

const Template: Story<ExperienceCardComponent> = (
  args: ExperienceCardComponent
) => ({
  props: {
    experience: args.experience,
  },
  template: `<rc-experience-card [experience]="experience"></rc-experience-card>`,
});

export const Default = Template.bind({});
Default.args = {
  experience: {
    client: 'CBP',
    localization: 'Saint-Herblain, 44800, France',
    roles: [
      {
        titre: 'Lead developpeur front-end',
        contractType: 'FREELANCE',
        startDate: '2021-01-01',
        endDate: null,
      },
    ],
    tasks: [
      'Mise en place une approche atomic design',
      'challengé / formé l’équipe aux développement Angular (présentations, pair programming)',
      'prôné l’usage du TDD',
      'challengé les maquettes fournies par l’UX',
      'participé aux recrutement pour diverses équipes',
      'travaillé de pair avec la PO afin de prioriser les tâches avec un coefficient de plus value',
      'remonté les alertes nécessaires sur l’avancée des devs',
      'POC pour la mise en place de Storybook',
      'POC pour la mise en place de Nx',
      'POC pour la mise en place de tests Cypress avec Storybook',
    ],
    technos: [
      'Nx',
      'Angular',
      'Jest',
      'Storybook',
      'Cypress',
      'Sass',
      'Angular material CDK',
      'ngx-translate',
      'Rxjs',
    ],
    description: 'Refonte d’une application de gestion de sinistres en Angular',
  },
};
