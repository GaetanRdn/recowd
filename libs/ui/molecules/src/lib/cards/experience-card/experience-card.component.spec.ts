import { TestBed, waitForAsync } from '@angular/core/testing';
import { ExperienceCardComponent } from './experience-card.component';
import { TemplateLookup } from '@recowd/test/utils';
import { Component } from '@angular/core';
import { CBP_EXPERIENCE } from '@recowd/test/data';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';

describe('ExperienceCardComponent', () => {
  let templateLookup: TemplateLookup<HostComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [HostComponent],
      imports: [ExperienceCardComponent, NoopAnimationsModule],
    }).compileComponents();

    templateLookup = new TemplateLookup(HostComponent);
    templateLookup.detectChanges();
  }));

  it('should create', () =>
    expect(templateLookup.firstChildElement).toMatchSnapshot());
});

@Component({
  template: ` <rc-experience-card
    [experience]="experience"
  ></rc-experience-card>`,
})
class HostComponent {
  public experience: ExperienceCardComponent['experience'] = CBP_EXPERIENCE;
}
