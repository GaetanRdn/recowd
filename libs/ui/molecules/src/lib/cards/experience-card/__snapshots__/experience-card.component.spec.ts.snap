// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperienceCardComponent should create 1`] = `
<rc-experience-card>
  <rc-card>
    <div
      class="rc-card-header"
    >
      <rc-card-title
        class="rc-card-title"
      >
        CBP
      </rc-card-title>
      <rc-card-subtitle
        class="rc-card-subtitle"
      >
        <span>
          2 an(s) et 7 mois
        </span>
        <span>
          Saint-Herblain
        </span>
      </rc-card-subtitle>
    </div>
    <rc-card-content
      class="rc-card-content"
    >
      <rc-accordion>
        <rc-accordion-bloc
          class="rc-accordion-bloc"
        >
          <rc-accordion-bloc-header
            class="rc-accordion-bloc-header"
          >
            <rc-roles-list>
              <rc-list
                role="list"
              >
                <span
                  class="rc-list-items-wrapper"
                >
                  <rc-list-item
                    class="rc-list-item"
                    role="list-item"
                  >
                    <rc-line
                      class="rc-line"
                    >
                      Lead developer
                    </rc-line>
                    <rc-line
                      class="rc-line"
                    >
                      FREELANCE
                    </rc-line>
                    <rc-line
                      class="rc-line"
                    >
                      January 2020 - July 2022
                    </rc-line>
                  </rc-list-item>
                  <rc-list-item
                    class="rc-list-item"
                    role="list-item"
                  >
                    <rc-line
                      class="rc-line"
                    >
                      Lead developer
                    </rc-line>
                    <rc-line
                      class="rc-line"
                    >
                      CDI ESN
                    </rc-line>
                    <rc-line
                      class="rc-line"
                    >
                      January 2020 - December 2020
                    </rc-line>
                  </rc-list-item>
                  
                </span>
              </rc-list>
            </rc-roles-list>
            <rc-material-icon
              class="ng-trigger ng-trigger-rotate material-icons"
              style=""
            >
               expand_more 
            </rc-material-icon>
          </rc-accordion-bloc-header>
          <rc-accordion-bloc-content
            class="ng-trigger ng-trigger-collapsable rc-collapsed"
            style="height:0px;opacity:0.5;0:height;1:opacity;"
          >
            <span
              class="rc-accordion-bloc-content-wrapper"
            >
              <p
                class="rc-quote"
                rcquote=""
              >
                Great things
              </p>
              <rc-list
                class="rc-horizontal"
                horizontal=""
                role="list"
              >
                <rc-list-title
                  class="rc-list-title label"
                >
                  Technos
                </rc-list-title>
                <span
                  class="rc-list-items-wrapper"
                >
                  <rc-list-item
                    class="rc-list-item"
                    role="list-item"
                    style=""
                  >
                    Angular
                  </rc-list-item>
                  
                </span>
              </rc-list>
              <rc-list
                class="tasks"
                role="list"
              >
                <rc-list-title
                  class="rc-list-title"
                >
                  Tâches effectuées
                </rc-list-title>
                <span
                  class="rc-list-items-wrapper"
                >
                  <rc-list-item
                    class="rc-list-item"
                    role="list-item"
                    style=""
                  >
                    many things
                  </rc-list-item>
                  
                </span>
              </rc-list>
            </span>
          </rc-accordion-bloc-content>
        </rc-accordion-bloc>
      </rc-accordion>
    </rc-card-content>
  </rc-card>
</rc-experience-card>
`;
